### PR TITLE
Overlay lighting if it was awesome (small refactor)

### PR DESCRIPTION
**Pull request by ChungusGamer666**
## About The Pull Request

Something that had been bugging me for a while is that overlay lights were not updating lumcount on the correct turfs - This is especially egregious with high luminosity eyes. 
So I had to do this small refactor to fix that.

But that is not all, I also added proper support for non cardinal directions for both beams and normal movable lights.
Can you believe we didn't have that, still?
image
image

## Why It's Good For The Game

Not updating the lumcount of turfs properly slightly broke a bunch of mechanics in the game, like nightmares.
This will fix quite a lot of these issues.
image
image

### CODE REVIEW

**Comment by ChungusGamer666** [comment]
i just need to fix a small oversight with range 0 beams and then this will be good to go

**Comment by ChungusGamer666** [comment]
there it is

File: code/datums/components/overlay_lighting.dm [hunk 1]
```diff
@@ -496,35 +501,72 @@
...
+		var/final_turf = get_step(scanning, current_direction) || scanning
+		for(var/turf/lit_turf in get_line(get_turf(current_holder), final_turf))
+			lit_turf.dynamic_lumcount += lum_power
+			. += lit_turf
+	else
+		for(var/turf/lit_turf in view(lumcount_range, scanning))
+			lit_turf.dynamic_lumcount += lum_power
+			. += lit_turf
+	if(length(.))
+		affected_turfs = .
```
Comment (san7890) [inline code/datums/components/overlay_lighting.dm:568]:
```suggestion
		affected_turfs = .
		
	return .
```

if you're gonna do funny dot operator stuff you must pay the dot operator toll

**Review by san7890 — CHANGES_REQUESTED** [review]
i have no insights on the rest of this code but there is one important thing

**Comment by ChungusGamer666** [comment]
@LemonInTheDark I believe you'll be very much interested in reviewing this

**Comment by ChungusGamer666** [comment]
please... merge this shit

**Comment by san7890** [comment]
i don't know if your account being nuked also fucked up how notifs work so you might have to reach out to him via the grapevine

**Comment by ChungusGamer666** [comment]
@LemonInTheDark hello sweetheart this pr should be visible now you really should review it because lighting seems to be your pet project

**Comment by ChungusGamer666** [comment]
the rot consumes

**Comment by LemonInTheDark** [comment]
I do actually want this am just a bum

**Pull request closed without merging**